test/test.make

```diff
@@ -0,0 +1,38 @@
+.SECONDEXPANSION:
+BUILDS= normal gc64 dualnum nojit
+
+normal : XCFLAGS= 
+gc64 : XCFLAGS=-DLUAJIT_ENABLE_GC64
+dualnum : XCFLAGS=-DLUAJIT_NUMMODE=2
+nojit : XCFLAGS=-DLUAJIT_DISABLE_JIT
+
+#%.build: ../src/*.h ../src/*.c
+builds/%/luajit builds/%/libluajit.so: ../src/*.h ../src/*.c
+	@echo "==== Building LuaJIT target = ${BUILD_TARGET} ===="
+	$(MAKE) -C ../src -j XCFLAGS="-DLUAJIT_ENABLE_LUA52COMPAT ${XCFLAGS}"
+	mkdir -p builds/${BUILD_TARGET}/
+	cp ../src/luajit builds/${BUILD_TARGET}/
+	cp ../src/libluajit.so builds/${BUILD_TARGET}/
+	$(MAKE) -C ../src clean
+	@echo "==== Successfully built LuaJIT ===="
+
+define make_buildtarget
+  $1_BINARIES= builds/$1/libluajit.so builds/$1/luajit
+  #$$($1_BINARIES): $1.build
+  $1: $$($1_BINARIES)
+  $1: BUILD_TARGET=$1
+endef
+
+$(foreach build, $(BUILDS), $(eval $(call make_buildtarget,$(build))))
+
+all : normal gc64 dualnum nojit

```

Comment:
Conventionally this should be first rule in the Makefile (since just typing `make` executes the first rule in the Makefile).